### DISCUSSION TITLE
arc_to_bulge question

### DISCUSSION THREAD

**Discussion by drmerc** [mozman/ezdxf#373] [Q&A] [answered]
I'm attempting to convert arcs into polylines using the arc_to_bulge function.
for example:
    for e in doc.query("ARC"):
        arc_point = ezdxf.math.arc_to_bulge(e.dxf.center, e.dxf.start_angle, e.dxf.end_angle, e.dxf.radius)

The start and end points of the arc_point don't match e.start_point and e.end_point and results in a larger arc than expected.  

Is there something I'm missing?

Thanks!

**Comment by Timerlen** [comment]
This code can be used to check the start point and end point of a circle 
from ezdxf.math import Vector
if b.dxftype() == "ARC":
    start_point = Vector.from_deg_angle(b.dxf.start_angle, b.dxf.radius)
    end_point = Vector.from_deg_angle(b.dxf.end_angle, b.dxf.radius)
    s = start_point + b.dxf.center
    e = end_point + b.dxf.center
Maybe something is shifted?

**Reply by mozman** [reply]
FYI: This works only if the extrusion vector == (0, 0, 1), which means the OCS == WCS. An extrusion vector != (0, 0, 1) can also appear if you only work in 2D. CAD applications can use the extrusion vector (0, 0, -1) to change the arc orientation from counter clockwise to clockwise orientation. I found this behavior for arcs in mirrored and then exploded block references.

**Reply by Timerlen** [reply]
Now I know that I know little. Is there perhaps more extensive documentation somewhere?

**Comment by mozman** [comment, marked as answer]
The DXF format store angles in degrees, the `arc_to_bulge()` function requires radians as input.

**Reply by drmerc** [reply]
That was it! Thank you
